PyDSS/pyPostprocessor/PostprocessScripts/AutomatedThermalUpgrade.py

```diff
@@ -1,1001 +1,1453 @@
+                    elif line_name in self.all_line_ldgs_alltps:
+                        self.all_line_ldgs_alltps[line_name][0].append(max(line_current))
+                if ldg > self.config["line_loading_limit"] and switch == "False":  # and switch==False:
+                    if line_name not in self.line_violations_alltps:
+                        self.line_violations_alltps[line_name] = [[max(line_current)], line_limit]
+                    elif line_name in self.line_violations_alltps:
+                        self.line_violations_alltps[line_name][0].append(max(line_current))
+                if not dss.ActiveClass.Next() > 0:
+                    break
+        return
+
+    # this function solves using list of tps_to_test
+    def solve_diff_tps_lines(self):
+        # Uses Kwami's LA100 logic
+        self.logger.info("PVsystems: %s",dss.PVsystems.Count())
+        self.line_violations_alltps = {}
+        self.all_line_ldgs_alltps = {}
+        if len(self.other_load_dss_files)>0:
+            for tp_cnt in range(len(self.config["tps_to_test"])):
+                # First two tps are for disabled PV case
+                if tp_cnt==0 or tp_cnt==1:
+                    dss.run_command("BatchEdit PVSystem..* Enabled=False")
+                    dss.Loads.First()
+                    while True:
+                        load_name = dss.Loads.Name().split(".")[0].lower()
+                        if load_name not in self.min_max_load_kw:
+                            print("Load not found, quitting...")
+                            quit()
+                        new_kw = self.min_max_load_kw[load_name][tp_cnt]
+                        dss.Loads.kW(new_kw)
+                        if not dss.Loads.Next() > 0:
+                            break
+                    self.dssSolver.Solve()
+                    if not dss.Solution.Converged():
+                        raise OpenDssConvergenceError("OpenDSS solution did not converge")
+                if tp_cnt==2 or tp_cnt==3:
+                    dss.run_command("BatchEdit PVSystem..* Enabled=True")
+                    dss.Loads.First()
+                    while True:
+                        load_name = dss.Loads.Name().split(".")[0].lower()
+                        if load_name not in self.min_max_load_kw:
+                            print("Load not found, quitting...")
+                            quit()
+                        new_kw = self.min_max_load_kw[load_name][tp_cnt - 2]
+                        dss.Loads.kW(new_kw)
+                        if not dss.Loads.Next() > 0:
+                            break
+                    self.dssSolver.Solve()
+                    self.logger.info("TotalPower=%s", dss.Circuit.TotalPower())
+                    if not dss.Solution.Converged():
+                        raise OpenDssConvergenceError("OpenDSS solution did not converge")
+                dss.Circuit.SetActiveClass("Line")
+                dss.ActiveClass.First()

```

Comment:
This code needs to check the return of `First()`.  If it returns 0 then a segfault will occur below.